nautobot/core/api/utils.py

```diff
@@ -171,17 +171,19 @@ def get_view_name(view, suffix=None):
     """
     Derive the view name from its associated model, if it has one. Fall back to DRF's built-in `get_view_name`.
     """
-    if hasattr(view, "queryset"):
+    name = getattr(view, "name", None)
+    if name is not None:
+        return view.name
+    elif hasattr(view, "queryset"):
         # Determine the model name from the queryset.
-        name = view.queryset.model._meta.verbose_name
+        if hasattr(view, "detail") and view.detail:
+            name = view.queryset.model._meta.verbose_name
+        else:
+            name = view.queryset.model._meta.verbose_name_plural
```

Comment:
Not blocking but it might be nice to add a unit test or two specifically exercising this function for various API views.

Reply:
I may need some help with that part. I looked at the DRF code to see how/if they unit tested the get_view_name function and there was nothing.

Reply:
I was thinking nothing more complex than just calling `get_view_name(SomeExistingAPIViewClass)` and asserting that it returns the expected string for a few such cases. Happy to help if needed though.

Reply:
Added the unit tests